common/djangoapps/static_replace/__init__.py

```diff
@@ -180,7 +180,7 @@ def replace_static_url(original, prefix, quote, rest):
             else:
                 # if not, then assume it's courseware specific content and then look in the
                 # Mongo-backed database
-                url = StaticContent.convert_legacy_static_url_with_course_id(rest, course_id)
+                url = StaticContent.get_canonicalized_asset_path(course_id, rest)
```

Comment:
I think it's fine to pass in. It makes some of the test logic simpler anyhow -- no need to mock anything or clean anything up. And it's very explicit, not relying on patching a global on startup.